src/parquet/column/properties.h

```diff
@@ -104,33 +126,86 @@ class PARQUET_EXPORT WriterProperties {
       return this;
     }
 
+    Builder* disable_dictionary() {
+      dictionary_enabled_ = false;
+      return this;
+    }
+
     Builder* dictionary_pagesize(int64_t dictionary_psize) {
       dictionary_pagesize_ = dictionary_psize;
       return this;
     }
 
+    Builder* dictionary_size_threshold(int64_t dictionary_sizet) {
+      dictionary_size_threshold_ = dictionary_sizet;
+      return this;
+    }
```

Comment:
I believe this parameter is the same as the dictionary_pagesize_ 

https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java#L46

If the dictionary size exceeds the dictionary page size, then the encoder reverts to plain encoding.